src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -106,6 +125,110 @@ export const handleEncryptedSubmission: RequestHandler = async (
     })
   }
 
+  // Validate encrypted submission
+  const { encryptedContent, responses } = req.body
+  const encryptedEncodingResult = await checkIsEncryptedEncoding(
+    encryptedContent,
+  )
+  if (encryptedEncodingResult.isErr()) {
+    logger.error({
+      message: 'Error verifying content has encrypted encoding.',
+      meta: logMeta,
+      error: encryptedEncodingResult.error,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      encryptedEncodingResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+
+  // Process encrypted submission
+  const processedResponsesResult = await getProcessedResponses(form, responses)
+  if (processedResponsesResult.isErr()) {
+    logger.error({
+      message: 'Error processing encrypted submission.',
+      meta: logMeta,
+      error: processedResponsesResult.error,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      processedResponsesResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+  const processedResponses = processedResponsesResult.value
+  // eslint-disable-next-line @typescript-eslint/no-extra-semi
+  ;(req.body as WithParsedResponses<
+    typeof req.body
+  >).parsedResponses = processedResponses
+  // Prevent downstream functions from using responses by deleting it.
+  delete (req.body as SetOptional<EncryptSubmissionBody, 'responses'>).responses
```

Comment:
Could you help to add a TODO that this segment of code should eventually be refactored in #1104 ?